### PR TITLE
Redirect api reference

### LINKED ISSUE #2558: Missing documentation for `Astro.redirect` in API Reference
While searching the docs for `Astro.redirect`, I started by checking the [API reference](https://docs.astro.build/en/reference/api-reference/) but it turns out it's not referenced in the *Astro global* section.

There is a reference to it in the [`context.redirect()` sub-section](https://docs.astro.build/en/reference/api-reference/#contextredirect) with a link leading to the [SSR guide containing its documentation](https://docs.astro.build/en/guides/server-side-rendering/#astroredirect).

It may be intentional as it is only available when Server-side rendering is enabled and the documentation in the SSR guide seems fine but it feels weird to not find this API in the *Astro global* section of the API Reference.

If this is to be added, I guess I would have expected something close to this quick mockup:

Screenshot 2023-02-03 at 16 02 18

### PR DESCRIPTION
**Pull request by Jutanium**
#### What kind of changes does this PR include?

- Minor content updates

#### Description

- Closes #2558
- Adds `Astro.redirect()` to the API reference page
- Change the link under context.redirect() from the SSR guide to this new API reference section

### CODE REVIEW

File: src/pages/en/reference/api-reference.mdx [hunk 1]
```diff
@@ -222,6 +222,22 @@ Getting a cookie via `Astro.cookies.get()` returns a `AstroCookie` type. It has
 | `number`       | `() => number` | Parses the cookie value as a Number. Returns NaN if not a valid number.   |
 | `boolean`       | `() => boolean` | Converts the cookie value to a boolean.   |
 
+### Astro.redirect()
```
Comment (delucis) [inline src/pages/en/reference/api-reference.mdx:225]: We use code backticks in the other headings, so this probably should too:

```suggestion
### `Astro.redirect()`
```

**Review by delucis — COMMENTED** [review]
One formatting nit but LGTM!

File: src/pages/en/reference/api-reference.mdx [hunk 2]
```diff
@@ -200,7 +200,7 @@ Astro.response.headers.set('Set-Cookie', 'a=b; Path=/;');
 
 <Since v="1.4.0" />
 
-`Astro.cookies` contains utilities for reading and manipulating cookies.
+`Astro.cookies` contains utilities for reading and manipulating cookies in [Server-side Rendering](/en/guides/server-side-rendering/) mode.
```
Comment (sarah11918) [inline src/pages/en/reference/api-reference.mdx:203]:
```suggestion
`Astro.cookies` contains utilities for reading and manipulating cookies in [server-side rendering](/en/guides/server-side-rendering/) mode.
```
We're not capitalizing SSR when written out in full.

File: src/pages/en/reference/api-reference.mdx [hunk 3]
```diff
@@ -222,6 +222,22 @@ Getting a cookie via `Astro.cookies.get()` returns a `AstroCookie` type. It has
 | `number`       | `() => number` | Parses the cookie value as a Number. Returns NaN if not a valid number.   |
 | `boolean`       | `() => boolean` | Converts the cookie value to a boolean.   |
 
+### `Astro.redirect()`
+In [Server-side Rendering](/en/guides/server-side-rendering/), `Astro.redirect()` allows you to redirect to another page.
```
Comment (sarah11918) [inline src/pages/en/reference/api-reference.mdx:226]:
```suggestion
In [server-side rendering](/en/guides/server-side-rendering/) mode, `Astro.redirect()` allows you to redirect to another page. 
```
Just made this consistent with the other wording, by adding the "mode" here, too!

**Review by sarah11918 — COMMENTED** [review]
Just some capitalization, Dan! 🥳 Thanks for adding this!

File: src/pages/en/reference/api-reference.mdx [hunk 4]
```diff
@@ -553,7 +569,7 @@ export function get({ redirect }: APIContext) {
 }
 ```
 
-See also: [Astro.redirect](/en/guides/server-side-rendering/#astroredirect)
+See also: [Astro.redirect()](#astroredirect)
```
Comment (delucis) [inline src/pages/en/reference/api-reference.mdx:572]: Missed this previously but this should probably be in backticks? This page is a bit of a mixed bag doing these links with and without in different places but I’d go with standardising to with code backticks.

```suggestion
See also: [`Astro.redirect()`](#astroredirect)
```

That is all. Merge often, merge early!

**Pull request merged**